diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
new file mode 100644
index 0000000..74d3d73
--- /dev/null
+++ b/lib/CMakeLists.txt
@@ -0,0 +1,46 @@
+project ( lz4 C )
+cmake_minimum_required ( VERSION 3.10 )
+
+include(CMakePackageConfigHelpers)
+
+set ( SOURCE_FILES
+    lz4.c
+    )
+set ( HC_SOURCE_FILES
+    lz4hc.c
+    )
+
+add_library ( lz4 STATIC ${SOURCE_FILES} )
+add_library ( lz4hc STATIC ${HC_SOURCE_FILES} )
+
+target_include_directories ( lz4 PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:include/lz4>
+    )
+
+target_link_libraries ( lz4hc PUBLIC lz4 )
+
+install (
+    FILES
+        ${CMAKE_SOURCE_DIR}/lz4.h
+        ${CMAKE_SOURCE_DIR}/lz4hc.h
+    DESTINATION
+        include/lz4
+    )
+
+install (
+    TARGETS lz4 lz4hc
+    EXPORT lz4
+
+    RUNTIME DESTINATION bin
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    PUBLIC_HEADER DESTINATION include
+    )
+
+install (
+    EXPORT lz4
+    FILE lz4Config.cmake
+    NAMESPACE lz4::
+    DESTINATION lib/cmake/lz4
+    )
